### ISSUE TITLE
forward port the webob httpexception changes to support json mime types

### ISSUE THREAD

**Issue by mmerickel** [Pylons/pyramid#2406]
https://github.com/Pylons/webob/pull/230

**Comment by bertjwregeer** [comment]
Taking this. Need to release WebOb 1.6.

**Comment by bertjwregeer** [comment]
If only Pyramid had lifted the top-level exception wholesale ;-)

**Comment by bertjwregeer** [comment]
So I just figured out that MIMEAccept in WebOb is busted as it is currently written.

Accept header:

`application/json, text/html, */*;q=0.8`

Server offers the following variants:

`text/plain, text/html, application/json`

with `text/plain` being the default if no offers match at all.

Current `MIMEAccept.best_match()` will return `text/html`, even though `application/json` is preferred by the client.

Fudging the numbers to `application/json;q=1.0, text/html;q=0.9, */*;q=0.8` which is exactly the same thing I expressed above, but implicitly rather than explicitly and `MIMEAccept.best_match()` now does the right thing.
